Automattic/woocommerce-payments
# Clean the code which checks for UPE enablement

Closes https://github.com/Automattic/woocommerce-payments/issues/7779

#### Changes proposed in this Pull Request
The main goal of this PR is to remove the outstanding flags used in the `class-wc-payments-features.php`. Since these flags and their values were heavily used across the admin panel in the pages like settings, notes, and modals, these pages were also cleaned up.

> 💡 The code for creating the UPE note was removed, but saved admin notes are kept in the merchant's site database and must be cleaned up. To analyse this further and perform a necessary notes cleanup, a separate issue will be created by @timur27.
UPDATE: see https://github.com/Automattic/woocommerce-payments/pull/7821#issuecomment-1849681366










#### Changes & testing instructions for them
* `enable-upe-preview` task https://github.com/Automattic/woocommerce-payments/pull/7821/commits/950037e624405f15a0e91b8741dad2d2ddfae15c because https://github.com/Automattic/woocommerce-payments/pull/6417/commits/3be27ba124e6d272be8528d5243edbb52cf9777b#diff-373a302d7bd472846a42de31047251187bf6182137a74cf2cf5b0eac87be246e and I don't see any areas where this task is being used
* https://github.com/Automattic/woocommerce-payments/pull/7821/commits/a5011e4e368a7f8e142310502d2e109e26b3155d
    * UPE context was used to share the flag state and the request status. Since none of them are relevant anymore, the context and its provider were removed
* https://github.com/Automattic/woocommerce-payments/pull/7821/commits/071af59a3523bfb6f141fd3b36f5a41ef90c16fa
    * Ensure that [the APM task](https://github.com/Automattic/woocommerce-payments/assets/22319444/c69ec0a1-0c22-40c8-a59d-22561208bdc8) is shown to the merchant when only card is enabled
    * Confirm that after clicking on this task, the merchant is redirected to the additional payment methods setup page
* https://github.com/Automattic/woocommerce-payments/pull/7821/commits/f37dd06688a037157442d55fa0b3bcea2c777a25
    * Perform the below two checks, at least in Blocks but if possible - also in shortcode
    * Disable WooPay, pay with a card & any additional payment method, and confirm that both payments are successful
    * Enable WooPay and perform the same check as in the previous bullet point
    * Only through blocks or only through shortcode, use WooPay to checkout and confirm the checkout is successful
* https://github.com/Automattic/woocommerce-payments/pull/7821/commits/0c7bb4c4c55829c9e792a6a857b75ea3f6fad6dd
    * confirm that payment elements are initialized on pay-for-order and add-payment-method pages
    * confirm that in shortcode checkout, terms are correctly disaplyed/removed when checking/unchecking the save payment method checkbox for cc
* https://github.com/Automattic/woocommerce-payments/pull/7821/commits/8d3c7503f35f9fdb1477524e8a3940e35618702b
    * Navigate to WooCommerce -> Settings and UPE field should disappear from WooPayments section while UPE Enabled APMs should be adjusted to be Enabled APMs now.
* https://github.com/Automattic/woocommerce-payments/pull/7821/commits/96a072d9d2494031d85c680df80ccb4806db7422
    * disable all APMs, remove EUR from the multi-currencies tab, go to Payments -> Overview and click on Add more ways for buyers to pay, add e.g. Bancontact, navigate back to multi-currencies and confirm EUR is added there
    * check the feedback modal for payment methods works correctly
* manual capture modal tests
    * navigate to Payments -> Settings, you should see that UPE methods can be enabled/disabled and appear just like the Credit card / Debit card method
    * check the "Issue an authorization on checkout, and capture later" checkbox
    * a confirmation modal is shown with the relevant text
    * if you cancel, then nothing has changed for the appearance of the payment methods
    * if you try again to check the checkbox and confirm your choice, you should now see that the appearance of UPE methods has changed
    * instead of a checkbox to enable/disable, there's a warning icon which shows a tooltip when hovering over it
    * if you uncheck the checkbox, you should not see a confirmation modal and the appearance should be back to normal for all payment methods


Besides the instructions mentioned above, please perform sanity check by placing an order through blocks and shortcode, while using cc and non-cc, with WooPay enabled and disabled.

-------------------


**Post merge**
